### ISSUE TITLE
Filtering with `scout -t` should roll back on <Escape>

### ISSUE THREAD

**Issue by FichteFoll** [ranger/ranger#900]
When filtering with <kbd>scout -t</kbd>, the file list view gets updated instantly when something is typed. However, when pressing <kbd>Enter</kbd> to cancel the console input, the filter remains.

#### ISSUE TYPE

- Improvement/feature request

#### RUNTIME ENVIRONMENT

- Operating system and version: Arch
- Terminal emulator and version: termite
- Python version: 3.6.1
- Ranger version/commit: 086074db6f08af058ffdced7319287715a88d42a
- Locale: en_US

#### EXPECTED BEHAVIOR

Undo or roll back the current filtering operation.

#### CURRENT BEHAVIOR

Filter remains.

**Comment by hut** [comment]
`:scout -t` alone does not change the filter at all, so I assume you use it in combination with `-f`. When I use `:scout -tf`, then type something to filter the files, and then type `ESC`, the filter is reset. So I can't reproduce it. Try with `ranger --clean`, if it works then the issue is your outdated configuration.

**Comment by FichteFoll** [comment]
You're right, it doesn't happen with `:scout -t`. I should have verified this, since I discovered it with `:filter` initially.

The actual flags needed are both `-t` amd `-p`.

**Comment by hut** [comment]
So yeah, it would make sense for `:filter foo<ESC>` to roll back the fliter.

**Comment by kevinhwang91** [comment]
https://github.com/ranger/ranger/pull/2116 Do you mean this feature?

**Comment by FichteFoll** [comment]
Yes, that looks like it should address my issue, according to the readme and the gif. I can't comment on the code as I am entirely unfamiliar with it.
